### DISCUSSION TITLE
Array Query Keys

### DISCUSSION THREAD

**Discussion by apenab** [TanStack/query#1279] [Q&A] [answered]
If I use an array as a query key, how to select this query if it's not a string?
How to use this code?
`const data = queryCache.getQueryData(queryKey)`

**Comment by Andrey-Bazhanov** [comment]
Just pass array in `getQueyData`. Check docs for more info https://react-query.tanstack.com/docs/guides/queries#query-keys

**Comment by apenab** [comment, marked as answer]
But, when a query key is a variable(prop)? And the component that use `getQueyData` is not int the same component tree.

**Reply by TkDodo** [reply]
We usually create a function that creates the key array for us:

```
export const buildArticleKey = (id) => ['articles', id]
```

then, we can just call that function to build a key if we have the id.
